desktop/cmp/treemap/TreeMap.js

```diff
@@ -0,0 +1,317 @@
+        const {data, highchartsConfig, algorithm, tooltip, maxNodes} = this.model,
+            {defaultTooltip} = this;
+
+        return {
+            ...highchartsConfig,
+            tooltip: {
+                enabled: !!tooltip,
+                useHTML: true,
+                padding: 0,
+                shape: 'square',
+                shadow: false,
+                pointFormatter: function() {
+                    if (!tooltip) return;
+                    const {record} = this;
+                    return isFunction(tooltip) ? tooltip(record) : defaultTooltip(record);
+                }
+            },
+            series: [{
+                data,
+                type: 'treemap',
+                animation: false,
+                layoutAlgorithm: algorithm,
+                borderWidth: 0,
+                turboThreshold: maxNodes,
+                dataLabels: {
+                    enabled: true,
+                    allowOverlap: false,
+                    align: 'left',
+                    verticalAlign: 'top',
+                    style: {textOutline: 'none', visibility: 'hidden'}
+                },
+                events: {click: this.onClick}
+            }]
+        };
+    }
+
+    //----------------------
+    // Click handling
+    //----------------------
+    onClick = (e) => {
+        this._clickCount++;
+        this._debouncedClickHandler(e.point.record, e);
+        if (this._clickCount >= 2) this._debouncedClickHandler.flush();
+    };
+
+    clickHandler(record, e) {
+        try {
+            const {onClick, onDoubleClick} = this.model;
+            if (onClick && this._clickCount === 1) {
+                onClick(record, e);
+            } else if (onDoubleClick) {
+                onDoubleClick(record, e);
+            }
+        } finally {
+            this._clickCount = 0;
+        }
+    }
+
+    //----------------------
+    // Selection handling
+    //----------------------
+    syncSelection() {
+        if (!this._chart) return;
+
+        const {selectedIds, maxDepth, gridModel, store} = this.model;
+
+        // Fallback to parent node if selection exceeds max depth
+        let toSelect;
+        if (maxDepth && gridModel && gridModel.treeMode) {
+            toSelect = new Set(selectedIds.map(id => {
+                const record = store.getById(id);
+                return record ? record.xhTreePath.slice(0, maxDepth).pop() : null;
+            }));
+        } else {
+            toSelect = new Set(selectedIds);
+        }
+
+        // Update selection in chart
+        this._chart.series[0].data.forEach(node => {
+            node.select(toSelect.has(node.id), true);
+        });
+    }
+
+    //----------------------
+    // Labels
+    //----------------------
+    async updateLabelVisibilityAsync() {
+        if (!this._chart) return;
+
+        // Show / hide labels by comparing label size to node size
+        await forEachAsync(this._chart.series[0].data, node => {
```

Comment:
I don't think we're getting anything from `forEachAsync()` here. Since we need to cap the series length anyway, these are not giant iterations.  Adding a debug tag for logging showed no waits, meaning all this is doing is adding an extra async layer without actually pausing to unblock.  I would remove in this case to simplify.